osf/management/commands/handle_duplicate_files.py

```diff
@@ -0,0 +1,262 @@
+        'to_remove': next_file['_id'],
+        'preserving': preserved_file['_id']
+    }
+    return info
+
+def return_basefilenode_values(file_queryset):
+    """
+    Returns an IncludeQuerySet that has the minimum values that we need to do
+    a file comparison
+    """
+    return file_queryset.values(
+        '_id',
+        'id',
+        'type',
+        'versions__location',
+        'versions__region_id',
+        'guids___id',
+        '_history',
+        'name',
+    )
+
+def compare_versions(first_file, second_file):
+    """
+    Compares version and _history information. If either one matches, assume the files are a match
+    """
+    versions_equal = compare_location_and_region(first_file, second_file)
+    history_equal = compare_history(first_file, second_file)
+
+    return versions_equal or history_equal
+
+def inspect_duplicates(file_summary):
+    """
+    Inspects duplicates of a particular filetype and determines how many need to be resolved manually.
+    Outputs an array of files that can be successfully deleted.
+    """
+    to_remove = []
+    error_counter = 0
+
+    for duplicate_record in file_summary:
+        safe_to_remove = True
+        target_id, name, parent_id, file_type, path, count = duplicate_record
+        target = AbstractNode.objects.get(id=target_id)
+
+        files = return_basefilenode_values(
+            BaseFileNode.objects.filter(
+                name=name,
+                type=file_type,
+                target_object_id=target_id,
+                parent_id=parent_id,
+                _path=path
+            ).order_by('created')
+        )
+
+        preserved_file = files.first()
+
+        for next_file in files.exclude(_id=preserved_file['_id']):
+            safe_to_remove = resources_match(preserved_file, next_file)
+            if safe_to_remove:
+                to_remove.append(stage_removal(preserved_file, next_file))
+
+        if not safe_to_remove:
+            error_counter += 1
+            logger.info('Contact admins to resolve: target: {}, name: {}'.format(target._id, name))
+
+    logger.info('{}/{} errors to manually resolve.'.format(error_counter, len(file_summary)))
+    logger.info('Files that can be deleted without issue:')
+    for entry in to_remove:
+        logger.info('{}'.format(entry))
+    return to_remove
+
+def remove_duplicates(duplicate_array, file_type):
+    """
+    :param duplicate_array, expecting an array of dictionaries of format
+        [{'guid_to_repoint': guid, 'to_remove': file__id, 'preserving': file__id}]
+    """
+    for duplicate in duplicate_array:
+        guid = Guid.load(duplicate['guid_to_repoint'])
+        preserving = BaseFileNode.objects.get(_id=duplicate['preserving'])
+        to_remove = BaseFileNode.objects.get(_id=duplicate['to_remove'])
+
+        if guid:
+            guid.referent = preserving
+            guid.save()
+
+        to_remove.delete()
+    logger.info('Duplicates removed of type {}'.format(file_type))
```

Comment:
Delete command reloads duplicates rather than using info already retrieved when locating duplicates in the first place.  This is speedy enough for files.  Folders I think should be deleted one a time, spread out.